cmd/cloud-gate/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/Symantec/Dominator/lib/log/serverlogger"
+	"github.com/Symantec/cloud-gate/broker"
+	"github.com/Symantec/cloud-gate/broker/aws"
+	"github.com/Symantec/cloud-gate/broker/configuration"
+	"github.com/Symantec/cloud-gate/broker/httpd"
+	"github.com/Symantec/tricorder/go/tricorder"
```

Comment:
What does `tricorder` dependency do? Is that for sending metrics to LMM?

Reply:
Sounds like over-complication and introducing of unnecessary dependencies :)

Reply:
If you want metrics, you're going to introduce dependencies. We definitely want metrics. The dependency added here (tricorder) is very lightweight.

Reply:
I'm not sure why that has to be the case — in CIS API we use LMM API directly, it's quite convenient.

Reply:
It's my firm opinion that _pushing_ metrics is a flawed architecture, as it leads to DDoS failures. This has been a persistent problem for LMM and has required more and more defenses to protect against it. When _polling_ metrics, there is a graceful degradation when the system has trouble keeping up, since the architecture is naturally rate limiting.

Reply:
But you mentioned earlier that Scotty *pushes* metrics to LMM, so it's still a push, but with an additional dependency and link in the chain.

Reply:
Scotty throttles the rate at which it pushes to other LMM components (it's officially part of LMM), based on how much LMM can take (measured and then configured). It's a single place where we can go and tune it, rather than having to tune thousands of endpoints, some of which may not even have opened up ports to allow control and/or haven't implemented rate limiting.

Reply:
Ah, I see. Thanks for the confirmation, that sounds good :+1: